src/vidpf.rs

```diff
@@ -366,8 +395,54 @@ impl VidpfKey {
     }
 }
 
+impl ConstantTimeEq for VidpfKey {
+    fn ct_eq(&self, other: &VidpfKey) -> Choice {
+        Choice::from(self.id)
+            .ct_eq(&Choice::from(other.id))
+            .bitand(self.value.ct_eq(&other.value))
+    }
+
+    fn ct_ne(&self, other: &VidpfKey) -> Choice {
+        self.ct_eq(other).not()
+    }
+}
+
+impl Encode for VidpfKey {
+    fn encode(&self, bytes: &mut Vec<u8>) -> Result<(), CodecError> {
+        bytes.extend_from_slice(&self.value[..]);
+        Ok(())
+    }
+
+    fn encoded_len(&self) -> Option<usize> {
+        Some(16)
+    }
+}
+
+impl ParameterizedDecode<bool> for VidpfKey {
+    fn decode_with_param(is_leader: &bool, bytes: &mut Cursor<&[u8]>) -> Result<Self, CodecError> {
+        let mut value = [0; 16];
+        bytes.read_exact(&mut value)?;
+        Ok(VidpfKey {
+            id: if *is_leader {
+                VidpfServerId::S0
+            } else {
+                VidpfServerId::S1
+            },
+            value,
+        })
+    }
+}
```

Comment:
footgun: I'm a little worried about this API betting misused down the line. The user might expect that the encoded VIDPF key includes the server ID, which it doesn't.

I think it would be a little safer to just inline this code where it's needed.